src/main/java/io/mosip/mimoto/controller/IssuersController.java

```diff
@@ -78,4 +92,73 @@ public ResponseEntity<Object> getIssuerConfig(@PathVariable("issuer-id") String
+            responseWrapper.setErrors(List.of(new ErrorDTO(API_NOT_ACCESSIBLE_EXCEPTION.getCode(), API_NOT_ACCESSIBLE_EXCEPTION.getMessage())));
+            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(responseWrapper);
+        }
+        responseWrapper.setResponse(credentialsSupported);
+
+        if (credentialsSupported.getSupportedCredentials().isEmpty()) {
+            logger.error("invalid issuer id passed - {}", issuerId);
+            responseWrapper.setErrors(List.of(new ErrorDTO(INVALID_ISSUER_ID_EXCEPTION.getCode(), INVALID_ISSUER_ID_EXCEPTION.getMessage())));
+            return ResponseEntity.status(HttpStatus.NOT_FOUND).body(responseWrapper);
+        }
+
+        return ResponseEntity.status(HttpStatus.OK).body(responseWrapper);
+    }
+
+    @GetMapping("/{issuer-id}/credentials/{credentials-supported-id}/download")
+    public ResponseEntity<?> generatePdfForVCCredentials(@RequestHeader("Bearer") String token,
+                                                         @PathVariable("issuer-id") String issuerId,
+                                                         @PathVariable("credentials-supported-id") String credentialsSupportedId) {
+
+        ResponseWrapper<Object> responseWrapper = new ResponseWrapper<>();
+        responseWrapper.setId(ID);
+        responseWrapper.setVersion("v1");
+        responseWrapper.setResponsetime(DateUtils.getRequestTimeString());
+
+        try{
+            IssuerDTO issuerConfig = issuersService.getIssuerConfig(issuerId);
+            CredentialIssuerWellKnownResponse credentialIssuerWellKnownResponse = issuersService.getCredentialWellKnownFromJson();
+            Optional<CredentialsSupportedResponse> credentialsSupportedResponse = credentialIssuerWellKnownResponse.getCredentials_supported().stream()
+                    .filter(credentialsSupported -> credentialsSupported.getId().equals(credentialsSupportedId))
+                    .findFirst();
+            if (credentialsSupportedResponse.isEmpty()){
+                logger.error("Invalid credential supported id passed - {}", credentialsSupportedId);
+                responseWrapper.setErrors(List.of(new ErrorDTO(INVALID_CREDENTIAL_SUPPORTED_TYPE_ID_EXCEPTION.getCode(), INVALID_CREDENTIAL_SUPPORTED_TYPE_ID_EXCEPTION.getMessage())));
+                return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(responseWrapper);
+            }
+            ByteArrayInputStream inputStream =  issuersService.generatePdfForVerifiableCredentials(token, issuerConfig, credentialsSupportedResponse.get(), credentialIssuerWellKnownResponse.getCredential_endpoint());
+            //PDF file name with issuer display name and credential supported type display name
+            String pdfFileName = issuerConfig.getDisplay().stream().filter(displayDTO -> displayDTO.getLanguage().equals("en")).map(DisplayDTO::getName).findFirst().orElse(null) +
```

Comment:
Extract `en` as default language constant

Reply:
For PDF Naming, we are using display.name  from issuers config + "_" + display.name from wellknown. 

But issuers config display is mocked from wellknown, so it will ideally be same. can we think for different name for PDF Name here  ?

Reply:
made it into a constant and referenced here